File: frontend/src/services/writing.ts
import axiosInstance from '@/axios'  // 修改这里，使用配置了认证的实例
import { Writing, WritingCreate, WritingUpdate } from '@/types/writing'

// 修改API路径，确保与后端路由匹配
const API_URL = '/api/writing/'  // 确保这里的路径与后端路由前缀匹配

export const getWritings = async (): Promise<Writing[]> => {
  const response = await axiosInstance.get(API_URL)  // 使用 axiosInstance
  return response.data
}

export const getWritingById = async (id: string): Promise<Writing> => {
  const response = await axiosInstance.get(`${API_URL}/${id}`)  // 使用 axiosInstance
  return response.data
}

export const createWriting = async (writing: WritingCreate): Promise<Writing> => {
  const response = await axiosInstance.post(API_URL, writing)  // 使用 axiosInstance
  return response.data
}

export const updateWriting = async (id: string, writing: WritingUpdate): Promise<Writing> => {
  try {
    console.log("更新写作:", id, writing);  // 添加调试日志
    const response = await axiosInstance.put(`${API_URL}${id}`, writing);
    return response.data;
  } catch (error) {
    console.error("更新写作失败:", error);
    throw error;
  }
}

export const deleteWriting = async (id: string): Promise<boolean> => {
  const response = await axiosInstance.delete(`${API_URL}/${id}`)  // 使用 axiosInstance
  return response.data
}

export const checkWriting = async (data: { content: string, topic: string }): Promise<any> => {
  try {
    console.log("发送写作检查请求:", data)  // 添加调试日志
    const response = await axiosInstance.post(`${API_URL}check`, data)  // 注意这里的路径
    console.log("收到写作检查响应:", response.data)  // 添加调试日志
    return response.data
  } catch (error) {
    console.error("写作检查请求失败:", error)
    throw error
  }
}

File: backend/app/api/routes/writing.py
from fastapi import APIRouter, Depends, HTTPException, status
from app.core.security import get_current_user
from app.models.user import UserInDB
from app.models.writing import Writing, WritingCreate, WritingUpdate
from app.services.writing import (
    get_writings_by_user, get_writing_by_id, create_writing, 
    update_writing, delete_writing, check_writing
)
from typing import List, Optional

# 确保创建了router实例
router = APIRouter()

# 确保路由路径正确
@router.get("/", response_model=List[Writing])
async def get_writings(current_user: UserInDB = Depends(get_current_user)):
    return get_writings_by_user(str(current_user.id))

@router.get("/{writing_id}", response_model=Writing)
async def get_writing(
    writing_id: str,
    current_user: UserInDB = Depends(get_current_user)
):
    writing = get_writing_by_id(writing_id, str(current_user.id))
    if not writing:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail="Writing not found"
        )
    return writing

@router.post("/", response_model=Writing)
async def create_writing_endpoint(
    writing: WritingCreate,
    current_user: UserInDB = Depends(get_current_user)
):
    return create_writing(writing, str(current_user.id))

@router.put("/{writing_id}", response_model=Writing)
async def update_writing_endpoint(
    writing_id: str,
    writing_update: WritingUpdate,
    current_user: UserInDB = Depends(get_current_user)
):
    updated_writing = update_writing(writing_id, writing_update, str(current_user.id))
    if not updated_writing:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail="Writing not found"
        )
    return updated_writing

@router.delete("/{writing_id}", response_model=bool)
async def delete_writing_endpoint(
    writing_id: str,
    current_user: UserInDB = Depends(get_current_user)
):
    success = delete_writing(writing_id, str(current_user.id))
    if not success:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail="Writing not found"
        )
    return success

@router.post("/check", response_model=dict)
async def check_writing_endpoint(
    data: dict,
    current_user: UserInDB = Depends(get_current_user)
):
    if "content" not in data or "topic" not in data:
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail="Content and topic are required"
        )
    
    return check_writing(data["content"], data["topic"])